### ISSUE TITLE
[Migration] About migrating to Azure

### ISSUE THREAD

**Issue by JordanVScher** [Microsoft/BotBuilder#4423]
## Bot Info
* SDK Platform: Node.js 
* SDK Version: 3.14
* Active Channels: Facebook
* Deployment Environment:  local development with Emulator/dev.botframework.com

I've been using my bot registered on dev.botframework.com for a while and now I have to migrate it to Azure. I have an Azure account but I have no subscription yet. The thing is I don't understand how the pricing of an Azure subscription works. Say I register my credit card number, I can use the bot for free for 30 days and then I will be charged? Or after that period I have to pay something (for how much and for how long?) to keep using the free services? If I have a server with an SSL certificate can I avoid using Azure entirely? There's also the issue of the bot in dev.botframework.com being registered on my personal e-mail and I have to migrate it to my company's azure account. 

I'm asking because my company is not sure about any of this and we want to now how much we have to pay to the keep our bot running or if we can avoid paying at all as we were doing.

**Comment by fanidamj** [comment]
You will need an Azure account to manage the Bot Channels Registration, which is available in free and premium tiers. You can create an Azure account with the same email you are registered for dev.botframework.com.  You can still use the free tier without cost. If you decide you want a SLA and increased messaging then you can move to premium tier. The full pricing details are available [here](https://azure.microsoft.com/en-us/pricing/details/bot-service/).

The bot registration (in Azure it's called "Bot Channel Registration") is still free - you only incur costs if you use the Azure Bot Service which includes bot hosting and storage account in addition to the bot registration. If you prefer to setup your own deployment environment, choose the Bot Channels Registration in Azure, and point it to your bot’s messaging endpoint —**all you need is a valid SSL certificate** (which you mentioned to have) for your bot’s messaging endpoint.
The Web App Bot & Functions Bot are provided as a convenience for those who want a streamlined dev/deploy experience. It also takes care of the SSL certificate.

Yes, you can migrate bot from your personal e-mail to your company' azure account. In order to that you have to add your company's account to the [Azure Active Directory](https://aad.portal.azure.com) and the go subscription on Azure portal and assign the role as Owner or contributor as per your requirement. Once the account has been added, when you click on the migration you will get the dropdown with options of the accounts associated with your account. You can select the one to which you want to migrate.

Please contact [Azure Support](https://azure.microsoft.com/en-us/support/options/) for any other billing related questions. Hope this helps!!

**Comment by fanidamj** [comment]
@JordanVScher Can you please send us the snapshot of the error. And did you create an Azure account with same email associate with dev.botframework.com?

**Comment by JordanVScher** [comment]
@fanidamj Thanks for the pricing clarification! It's exactly what I needed to confirm. :)  
As I said, the location bar at bot registration was gray and not offering any options, so I couldn't finish the process. The problem solved itself after 40 minutes although the automatic migration was giving me the same mistake and didn't start working until much later. Bot migration seems to be working now, but I registered a new bot and moved all the information from the old bot to the new one. It's working fine in the web chat and the bot emulator but not too smoothly on Messenger(It seems that options requests aren't working).
